portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,15 +42,18 @@
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        Query find = new Query("comment").addSort("timestamp", SortDirection.DESCENDING);
+        Query find = new Query("comment").addSort("date", SortDirection.DESCENDING);
         DatastoreService data = DatastoreServiceFactory.getDatastoreService();
         PreparedQuery results = data.prepare(find);
 
         //Construct an array of comment history
-        List<String> history = new ArrayList<>();
+        List<Comment> history = new ArrayList<>();
         for (Entity entry: results.asIterable()) {
             String curr = (String) entry.getProperty("comment");
-            history.add(curr);
+            Date time = (Date) entry.getProperty("date");
+            String user = (String) entry.getProperty("email");
+
+            history.add(new Comment(curr, time, user));
```

Comment:
Small nitpick on variable names. Try to be a little more verbose and descriptive. It can go a long way in making the code easier to read. E.g. maybe use `currentComment` instead of `curr` and `commentHistory` instead of `history`.